api/request_source/request_source_plugin.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+
+package nighthawk.request_source;
+
+import "google/protobuf/wrappers.proto";
+import "validate/validate.proto";
+import "api/client/options.proto";
+
+// Configuration for DummyPluginRequestSource (plugin name: "nighthawk.dummy-request-source-plugin")
```

Comment:
I think we could use better documentation outlining what these individual plugin types do. This being the public API might be a good place to add a few sentences next to each plugin. That documentation should help the reader understand what the plugin is for. The current documentation is fairly short and requires the reader to already have some context.

Reply:
Understood. I hope that I've added some context for this.